lib/builders/documentTypes/documentTypeBuilder.ts

```diff
@@ -139,43 +133,48 @@ export class DocumentTypeBuilder {
   }
 
   build() {
-    if (this.id == null) {
+    if (!this.id) {
       const crypto = require('crypto');
       this.id = crypto.randomUUID();
     }
 
+    this.defaultTemplateObject = this.defaultTemplateId ? {id: this.defaultTemplateId} : null;
+    this.collectionObject = this.collectionId ? {id: this.collectionId} : null;
+    this.folderObject = this.folderId ? {id: this.folderId} : null;
+
     return {
-      alias: this.alias,
-      name: this.name,
+      alias: this.alias || "",
+      name: this.name || "",
       description: this.description || "",
       icon: this.icon || "icon-document",
       allowedAsRoot: this.allowedAsRoot || false,
       variesByCulture: this.variesByCulture || false,
       variesBySegment: this.variesBySegment || false,
+      collection: this.collectionObject || null,
```

Comment:
I'd like to offer you an alternative solution:
```suggestion
      collection: this.collectionId ? {id: this.collectionId} : null,
```
With this solution, you won't need to use the collectionObject variable. The same applies to the defaultTemplate and folder fields.